contracts/mesh-provider/src/ibc.rs

```diff
@@ -213,33 +223,110 @@ pub fn ack_list_validators(
     for val in res.validators {
         VALIDATORS.save(deps.storage, &val, &Validator::new())?;
     }
+
+    // reset retry counter
+    let mut config = CONFIG.load(deps.storage)?;
+    config.list_validators_retries_remaining = LIST_VALIDATORS_RETRIES;
+    CONFIG.save(deps.storage, &config)?;
+
     Ok(IbcBasicResponse::new())
 }
 
-pub fn fail_list_validators(_deps: DepsMut) -> Result<IbcBasicResponse, ContractError> {
-    // TODO: send another ListValidators message
-    unimplemented!();
+pub fn fail_list_validators(deps: DepsMut, env: Env) -> Result<IbcBasicResponse, ContractError> {
+    // check if we should retry
+    let cfg = CONFIG.load(deps.storage)?;
+    let channel_id = CHANNEL.load(deps.storage)?;
+    if cfg.list_validators_retries_remaining <= 0 {
+        return Err(ContractError::NoResponse("list_validators".to_string(), channel_id))
```

Comment:
An error here (in ack) will abort the receipt of the acknowledgement packet. This is usually not what we want to do.
We can let it come and record this as a failure somehow. Emitting a custom event is a good idea.

I would not just `add_attribute`, but rather `add_event` with a custom type just for these failure modes, for easy searching in the UI.